active/0000-qunit-dom.md

```diff
@@ -0,0 +1,105 @@
+- Start Date: 2018-02-12
+- RFC PR: (leave this empty)
+
+# Summary
+
+Introduce [`qunit-dom`] as a dependency by default in the `app` and `addon` blueprints.
+
+[`qunit-dom`]: https://github.com/simplabs/qunit-dom
+
+# Motivation
+
+> Why are we doing this?
+
+QUnit assertions are quite verbose and can look like this:
+
+```js
+assert.equal(this.element.querySelector('.title').textContent.trim(), 'Hello World!');
+```
+
+with `qunit-dom` we can write much more readable assertion for DOM elements:
+
+```js
+assert.dom('.title').hasText('Hello World!');
+```
+
+> What use cases does it support?
+
+It supports the most common assertions on DOM elements, like:
+
+- what text does the element have?
+- what value does the `<input>` element have?
+- is a certain CSS class applied to the element
+
+The full API is documented at <https://github.com/simplabs/qunit-dom/blob/master/API.md>.
+
+> What is the expected outcome?
+
+Using `qunit-dom` will lead to more simple and readable test code.
+
+
+# Detailed design
+
+The necessary changes to `ember-cli` are relatively small since we only need
+to add the dependency to the `app` blueprint, and the `addon` blueprint will
+inherit it automatically.
```

Comment:
Would it make sense to update the blueprints used for generating tests as well as e.g. a component integration test generated with `ember g …` currently generates this:

```js
assert.equal(this.element.textContent.trim(), 'template block text');
```

which (I think) should look like this with qunit-dom

```js
assert.dom(this.element).hasText('template block text');
```

Without actually updating the blueprints, I'm not sure anyone would even notice qunit-dom is even part of their project and enables nicer tests?